frida-build/src/lib.rs

```diff
@@ -37,37 +43,33 @@ pub fn download_and_use_devkit(kind: &str, version: &str) {
                 kind
             );
             // Download devkit
-            Command::new("wget")
-                .arg("-c")
-                .arg(format!(
-                        "https://github.com/frida/frida/releases/download/{}/frida-{}-devkit-{}-{}-{}.tar.xz",
-                        version,
-                        kind,
-                        version,
-                        env::var("CARGO_CFG_TARGET_OS").unwrap(),
-                        target_arch))
-                .arg("-O")
-                .arg(&devkit_tar)
-                .status()
-                .unwrap();
-        }
-        Command::new("tar")
-            .current_dir(&out_dir_path)
-            .arg("-xvf")
-            .arg(&devkit_tar)
-            .status()
-            .unwrap();
-        Command::new("mv")
-            .current_dir(&out_dir_path)
-            .arg(format!("libfrida-{}.a", kind))
-            .arg(format!(
-                "libfrida-{}-{}-{}.a",
+            let mut resp = reqwest::blocking::get(format!(
+                "https://github.com/frida/frida/releases/download/{}/frida-{}-devkit-{}-{}-{}.tar.xz",
+                version,
                 kind,
+                version,
                 env::var("CARGO_CFG_TARGET_OS").unwrap(),
-                target_arch
-            ))
-            .status()
-            .unwrap();
+                target_arch)).expect("devkit download request failed");
+            let mut out = File::create(&devkit_tar).expect("failed to create devkit tar file");
+            io::copy(&mut resp, &mut out).expect("failed to copy devkit tar content");
+        }
+        let tar_xz = File::open(&devkit_tar).expect("failed to open devkit tar.xz for extraction");
+        let tar = XzDecoder::new(tar_xz);
+        let mut archive = Archive::new(tar);
+        archive
+            .unpack(&out_dir_path)
+            .expect("cannot extract the devkit tar.gz");
+
+        let mut src_a = PathBuf::from(&out_dir_path);
+        src_a.push(format!("libfrida-{}.a", kind));
```

Comment:
For Windows, theese won't be `.a`, right?

Reply:
Will try to get this to run on a win machine now

Reply:
Right, on Windows it will be `.lib` not `.a`: https://github.com/frida/frida/releases/download/14.2.17/frida-gum-devkit-14.2.17-windows-x86_64.tar.xz